process/executionTrack/executionResultsTracker.go

```diff
@@ -0,0 +1,77 @@
+package executionTrack
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/data/block"
+)
+
+type executionResultConfirmed struct {
+	executionResult *block.ExecutionResult
+	confirmed       bool
+}
+
+type executionResultsTracker struct {
+	mutex            sync.RWMutex
+	executionResults map[string]*executionResultConfirmed
```

Comment:
I think we need to also track somehow if there are gaps, as we can only notarize the execution results in the order of the proposed blocks.

Reply:
refactored